packages/@lwc/engine/src/framework/component.ts

```diff
@@ -77,12 +77,12 @@ export function createComponent(uninitializedVm: UninitializedVM, Ctor: Componen
 
 export function getTemplateReactiveObserver(vm: VM): ReactiveObserver {
     return new ReactiveObserver(() => {
-        if (process.env.NODE_ENV !== 'production') {
-            assert.invariant(
-                !isRendering,
-                `Mutating property is not allowed during the rendering life-cycle of ${vmBeingRendered}.`
-            );
-        }
+        // if (process.env.NODE_ENV !== 'production') {
```

Comment:
if we remove the if condition around `valueMutated`, this check will need to be removed.